tests/suite/graceful_recovery_test.go

```diff
@@ -0,0 +1,316 @@
+	}
+
+	if err == nil {
+		return errors.New("expected request to error")
+	}
+
+	return nil
+}
+
+func checkContainerLogsForErrors(ngfPodName string) {
+	sinceSeconds := int64(15)
+	logs, err := resourceManager.GetPodLogs(
+		ngfNamespace,
+		ngfPodName,
+		&core.PodLogOptions{Container: nginxContainerName, SinceSeconds: &sinceSeconds},
+	)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(logs).ToNot(ContainSubstring("emerg"), logs)
+
+	logs, err = resourceManager.GetPodLogs(
+		ngfNamespace,
+		ngfPodName,
+		&core.PodLogOptions{Container: ngfContainerName, SinceSeconds: &sinceSeconds},
+	)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(logs).ToNot(ContainSubstring("error"), logs)
+}
+
+func waitForLeaderLeaseToChange(originalLeaseName string) error {
+	leaseCtx, leaseCancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer leaseCancel()
+
+	//nolint:nilerr
+	return wait.PollUntilContextCancel(
+		leaseCtx,
+		500*time.Millisecond,
+		true, /* poll immediately */
+		func(_ context.Context) (bool, error) {
+			leaseName, err := getLeaderElectionLeaseHolderName()
+			if err != nil {
+				return false, nil
+			}
+
+			if originalLeaseName != leaseName {
+				return true, nil
+			}
+
+			return false, nil
+		},
+	)
+}
+
+func getLeaderElectionLeaseHolderName() (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+	defer cancel()
+
+	var lease coordination.Lease
+	key := types.NamespacedName{Name: "ngf-test-nginx-gateway-fabric-leader-election", Namespace: ngfNamespace}
+
+	if err := k8sClient.Get(ctx, key, &lease); err != nil {
+		return "", errors.New("could not retrieve leader election lease")
+	}
+	return *lease.Spec.HolderIdentity, nil
+}
+
+func getContainerRestartCount(containerName, ngfPodName string) (int, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+	defer cancel()
+
+	var ngfPod core.Pod
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: ngfNamespace, Name: ngfPodName}, &ngfPod); err != nil {
+		return 0, errors.New("could not retrieve ngfPod")
+	}
+
+	var restartCount int
+	for _, containerStatus := range ngfPod.Status.ContainerStatuses {
+		if containerStatus.Name == containerName {
+			restartCount = int(containerStatus.RestartCount)
+		}
+	}
+
+	return restartCount, nil
+}
+
+func runNodeDebuggerJob(ngfPodName, jobScript string) (*v1.Job, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+	defer cancel()
+
+	var ngfPod core.Pod
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: ngfNamespace, Name: ngfPodName}, &ngfPod); err != nil {
+		return nil, errors.New("could not retrieve ngfPod")
+	}
+
+	b, err := resourceManager.GetFileContents("graceful-recovery/node-debugger-job.yaml")
+	if err != nil {
+		return nil, errors.New("error processing node debugger job file")
+	}
+
+	job := &v1.Job{}
+	_ = v1.AddToScheme(resourceManager.K8sClient.Scheme())
```

Comment:
we're changing the scheme of a global object. We should not do it in some specific test function. A good place for it is here https://github.com/nginxinc/nginx-gateway-fabric/blob/c7fd089c5d27e1eae8abe4e0871af6d5cac6335c/tests/suite/system_suite_test.go#L89